### PR TITLE
scheduled model-service polling to get modified detectors and deletin…

### PR DESCRIPTION
**Pull request by shsethi**
…g them from cachedDetectors

### CODE REVIEW

**Comment by williewheeler** [comment]
Please add unit tests

File: anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/DetectorManager.java [hunk 1]
```diff
@@ -37,13 +38,33 @@
...
+
+    private void initScheduler() {
+        scheduler.scheduleWithFixedDelay(() -> {
+            try {
+                this.detectorMapRefresh();
+            } catch (Exception e) {
+                log.error("Error refreshing detectors", e);
+            }
+        }, 1, detectorRefreshTimePeriod, TimeUnit.MINUTES);
+    }
```
Comment (williewheeler) [inline anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/DetectorManager.java]: If we get an exception, do we just lose the refresh until we restart the manager? What kind of exception are we expecting here, and is it something that we can recover from without losing refreshes entirely?

File: anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/DetectorManager.java [hunk 2]
```diff
@@ -96,4 +117,21 @@ private AnomalyDetector detectorFor(MappedMetricData mappedMetricData) {
...
+     */
+    List<UUID> detectorMapRefresh() {
+
+        List<UUID> updatedDetectors = new ArrayList<>();
+        detectorSource.findUpdatedDetectors(detectorRefreshTimePeriod).forEach(key -> {
+            updatedDetectors.add(key);
+            cachedDetectors.remove(key);
+        });
+
+        log.info("Removed detectors on refresh : {}"+updatedDetectors);
```
Comment (williewheeler) [inline anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/DetectorManager.java]: The `+` should be a `,`.

File: anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/util/DetectorResource.java [hunk 3]
```diff
@@ -30,4 +30,5 @@
 public class DetectorResource {
     private String uuid;
     private ModelTypeResource type;
+    private Boolean enabled;
```
Comment (williewheeler) [inline anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/util/DetectorResource.java]: Can you clarify what this is for, and how it gets set?

File: anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/util/ModelServiceConnector.java [hunk 4]
```diff
@@ -146,4 +150,29 @@ public ModelResource findLatestModel(UUID detectorUuid) {
 
         return modelResourceList.get(0);
     }
+
+    public DetectorResources findUpdatedDetectors(int timePeriod) {
+        isTrue(timePeriod > 0, "timePeriod can't be negative");
```
Comment (williewheeler) [inline anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/util/ModelServiceConnector.java]: Consider "timePeriod must be strictly positive". The current error message allows 0, but in fact that's not allowed.

File: anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/util/DetectorResource.java (hunk 3 above)
Comment (shsethi) [reply anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/util/DetectorResource.java]: Detectors are get disabled through and endpoint exposed by model-service, default is true, it's set to false when someones deletes detector from UI

File: anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/DetectorManager.java (hunk 1 above)
Comment (shsethi) [reply anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/DetectorManager.java]: This is to catch DetectorException's and any other Null pointer exceptions that might occur downstream. We don't loose the refresh when exception is caught.

**Pull request merged**